### PR TITLE
Create struct output for max and mean topology height

### PR DESCRIPTION
**Pull request by RShaw026**
## Description and Context

The function `ComputeMaxAndMean` has been changed, adhering to better coding practices. 

## Related Issues and Pull Requests

* Closes
* Blocks
* Is blocked by
* Follows
* Precedes
* Related to
* Part of
* Composed of

## How Has This Been Tested?

`ctest` is working. Building successfully with BACI, and the corresponding tests are passing in my local workstation.

## Checklist

- [ ] My commit messages mention the appropriate GitHub issue numbers and are longer than one line where appropriate.
- [ ] I have added/updated documentation where necessary.

## Interested Parties / Possible Reviewers

@mayrmt

### CODE REVIEW

File: src/mirco_topologyutilities.h [hunk 1]
```diff
@@ -48,15 +48,18 @@ namespace MIRCO
...
    * @param topology Topology matrix containing heights
-   * @param zmax Maximum height
-   * @param zmean Mean height
    */
-  void ComputeMaxAndMean(
-      Teuchos::SerialDenseMatrix<int, double> topology, double& zmax, double& zmean);
+  TopologyMaxAndMean ComputeMaxAndMean(const Teuchos::SerialDenseMatrix<int, double>& topology);
 }  // namespace MIRCO
 
 #endif  // SRC_TOPOLOGYUTILITIES_H_
```
Comment (isteinbrecher) [inline src/mirco_topologyutilities.h:65]: Newline is missing at the end of file

File: src/mirco_topologyutilities.cpp [hunk 2]
```diff
@@ -50,4 +52,5 @@ void MIRCO::ComputeMaxAndMean(
   }
 
   zmean = zmean / (topology.numCols() * topology.numRows());
+  return TopologyMaxAndMean{zmax, zmean};
 }
```
Comment (isteinbrecher) [inline src/mirco_topologyutilities.cpp:56]: Newline is missing at the end of file

File: src/mirco_topologyutilities.h [hunk 3]
```diff
@@ -48,15 +48,18 @@ namespace MIRCO
       bool RandomSeedFlag, std::string TopologyFilePath, bool RandomTopologyFlag,
       int RandomGeneratorSeed, Teuchos::RCP<TopologyGeneration>& surfacegenerator);
```
Comment (isteinbrecher) [inline src/mirco_topologyutilities.h:50]: Documentation is missing

Comment (RShaw026) [reply src/mirco_topologyutilities.h:50]: @isteinbrecher I have made the changes. Please review.

File: src/mirco_topologyutilities.h [hunk 4]
```diff
@@ -49,14 +49,24 @@ namespace MIRCO
...
+  struct TopologyMaxAndMean
+  {
+    double max_;
+    double mean_;
+  };
+
+  /**
+   * @brief Compute the maximum height and the mean height of the topology.
    *
    * @param topology Topology matrix containing heights
```
Comment (isteinbrecher) [inline src/mirco_topologyutilities.h:67]: The return documentation is missing here

File: src/mirco_topologyutilities.h [hunk 5]
```diff
@@ -49,14 +49,24 @@ namespace MIRCO
       int RandomGeneratorSeed, Teuchos::RCP<TopologyGeneration>& surfacegenerator);
 
   /**
-   * @brief Compute the maximum height (zmax) and the mean height (zmean) of the topology.
+   * @brief Store maximum and mean height of the topology
+   *
+   * @param max_ Maximum height of the topology
+   * @param mean_ Mean height of the topology
```
Comment (isteinbrecher) [inline src/mirco_topologyutilities.h:55]: In doxygen `@param` refers to the parameters of a function. In this case we mean member variables of the structure. You can either document them below, or don't document them individually as the naming is quite clear in this case.

**Review by isteinbrecher — APPROVED** [review]

**Pull request merged**